tests/components/calendar/test_trigger.py

```diff
@@ -85,8 +86,13 @@ async def async_get_events(
         """Get all events in a specific time frame, used by the demo calendar."""
         assert start_date < end_date
         values = []
+        local_start_date = dt_util.as_local(start_date)
```

Comment:
Note for reviewer: The motivation for this section is the test harness was incorrectly comparing all day events. This was just a test improvement, and is here as we've added more test coverage to more exhaustively test different types of calendar payloads